ruby_programming/test_driven_development/rspec_introduction_lesson.md

```diff
@@ -1,26 +1,245 @@
 
 ### Learning Outcomes
-*Look through these now and then use them to test yourself after doing the assignment*
 
-what the student is expected to know or be able to do by the end of this lesson
+*Look through these now and use them to guide your learning. By the end of this lesson, expect to:*
+
+- Know what RSpec is
+- Know how to install RSpec
+- Understand basic RSpec syntax (`describe` / `it`)
+
+### Introduction to RSpec
+
+#### What is RSpec, and why RSpec?
+
+At the most basic level, RSpec is a Ruby-written Domain Specific Language (DSL), or for the rest of us, a language specialized for a particular task. In this case, the task is testing Ruby code. The `rspec` gem comes packaged with all you need to get started, including at three gems: `rspec-core`, `rspec-expectations`, and `rspec-mocks`.
+
+At this point, you may be wondering, *Why RSpec? Surely, there are other frameworks out there*, and you'd be right. There are. In fact, at one point, Ruby came bundled with [Test::Unit]((https://github.com/test-unit/test-unit)) and later [Minitest](https://github.com/seattlerb/minitest) as part of its standard library, the latter of which lives on in Rails. If you tend to be pessimistic (I'm sorry, I meant *realistic*), then [Wrong](https://github.com/sconover/wrong) might be your cup of tea. Or perhaps you're hungry and in the mood for something more substantial, in which case a side plate of [Bacon](https://github.com/chneukirchen/bacon) might be what you need. At the end of the day, it doesn't matter which framework you choose as long as you cultivate your testing skills. 
+
+The reason we choose RSpec is ***readability***. The simplicity of this choice, however, is a proxy for a deeper goal. Because RSpec is foolishly easy to read and write, it's easy to pick-up and the intentions and meanings of your tests are immediately clear (aka, ***transparent***). RSpec's readability and transparency, in turn, lend themselves to ***maintainability***. Code that is readable and transparent is easier to maintain, and a testing suite that is equally so ensures that your application as whole remains adaptable to inevitable change.
+
+But enough proselytizing. Strap your helmet and buckle up; we're going to jump right in.
+
+#### Installing RSpec
+
+Boot up your terminal and punch in `gem install rspec` to install RSpec. Once that's done, you can verify your version of RSpec with `rspec --version`, which will output the current version of each of the packaged gems. Take a minute also to hit `rspec --help` and look through the various options available. We'll come back to those later.
+
+Finally, `cd` into a project directory that you wish to configure for use with RSpec and type `rspec --init` to initialize RSpec within the project. This will generate `.rspec` and `spec/spec_helper.rb`, such that your project might look like:
```

Comment:
Should we explain what `.rspec` and `spec/spec_helper.rb` are for? I'm not bothered either way, I just remember being confused about what the spec_helper was for when I was first learning rspec and wonder if other people will have the same experience.

Reply:
Yes! I thought about this as well and was wondering if I should include a section on configuring RSpec, but also thought that it might bog down the intro lesson.

I can try fitting it in, or perhaps we can spin off a smaller lesson or find an appropriate link in "Additional Resources"

Reply:
> it might bog down the intro lesson

Excellent point. Something in the additional resources would be perfect if you can find anything. If theres nothing out there this will be fine. We can always put something together if people ask about it in the chat.